File: core/commands/registry/irules/dosl7__is_ip_slowdown.py
# Enriched from F5 iRules reference documentation.
"""DOSL7::is_ip_slowdown -- Returns TRUE if source IP exists in greylist table"""

from __future__ import annotations

from ....compiler.side_effects import ConnectionSide, SideEffect, SideEffectTarget
from .._base import CommandDef
from ..models import CommandSpec, FormKind, FormSpec, HoverSnippet, ValidationSpec
from ..namespace_models import EventRequires
from ..signatures import Arity
from ._base import _IRULES_ONLY, register

_SOURCE = "https://clouddocs.f5.com/api/irules/DOSL7__is_ip_slowdown.html"


@register
class Dosl7IsIpSlowdownCommand(CommandDef):
    name = "DOSL7::is_ip_slowdown"

    @classmethod
    def spec(cls) -> CommandSpec:
        return CommandSpec(
            name="DOSL7::is_ip_slowdown",
            dialects=_IRULES_ONLY,
            hover=HoverSnippet(
                summary="Returns TRUE if source IP exists in greylist table",
                synopsis=("DOSL7::is_ip_slowdown",),
                snippet="Returns TRUE if source IP exists in greylist table",
                source=_SOURCE,
            ),
            forms=(
                FormSpec(
                    kind=FormKind.DEFAULT,
                    synopsis="DOSL7::is_ip_slowdown",
                ),
            ),
            validation=ValidationSpec(
                arity=Arity(),
            ),
            event_requires=EventRequires(),
            side_effect_hints=(
                SideEffect(
                    target=SideEffectTarget.DOSL7_STATE,
                    reads=True,
                    connection_side=ConnectionSide.BOTH,
                ),
            ),
        )

File: rust/tcl-registry/src/commands/irules/dosl7__is_ip_slowdown.rs
// tcl-lsp — a language server and toolchain for Tcl
// Copyright (C) 2026 James Deucker (bitwisecook) <https://github.com/bitwisecook>
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//
// SPDX-License-Identifier: AGPL-3.0-or-later

//! `DOSL7::is_ip_slowdown` iRules command.
use crate::prelude::*;
pub const fn spec() -> CommandSpec {
    CommandSpec {
        name: "DOSL7::is_ip_slowdown",
        dialects: Some(DialectSet::IRULES),
        arity: Arity::at_least(0),
        hover: Some(HoverSnippet {
            summary: "Returns TRUE if source IP exists in greylist table",
            synopsis: &["DOSL7::is_ip_slowdown"],
            snippet: "Returns TRUE if source IP exists in greylist table",
            source: "https://clouddocs.f5.com/api/irules/DOSL7__is_ip_slowdown.html",
            examples: "",
            return_value: "",
        }),
        forms: &[FormSpec {
            synopsis: "DOSL7::is_ip_slowdown",
            ..FormSpec::DEFAULT
        }],
        side_effects: &[SideEffect {
            target: SideEffectTarget::Dosl7State,
            reads: true,
            connection_side: ConnectionSide::Both,
            ..SideEffect::DEFAULT
        }],
        ..CommandSpec::DEFAULT
    }
}
